### ISSUE TITLE
`always-multi-line` allows newline even for non-multi-line values

### ISSUE THREAD

**Issue by ArmorDarks** [stylelint/stylelint#2760]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug.

It seems that when `declaration-colon-newline-after` set to `always-multi-line`, it suddenly allows to write in multiline style even properties with single value, without a list.

Refer to example below.

> Which rule, if any, is this issue related to?

`declaration-colon-newline-after` with `always-multi-line` as value

> What CSS is needed to reproduce this issue?

```scss
@font-face {
  src: // <- should be a way to disallow it
    something;
  src:
    something,
    something,
    something;
}
```

> What stylelint configuration is needed to reproduce this issue?

```yaml
  declaration-colon-newline-after: 'always-multi-line'
```

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config myconfig *.scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nope

> What did you expect to happen?

Ability to disallow newline after colon for specific cases.

**Comment by jeddy3** [comment]
I believe this rule is behaving as intended. It has a [very narrow](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#criteria-for-inclusion) scope. If a declaration spans multi-line lines, then a newline should always follow the colon. 

For example, the following is allowed:

```css
a {
  box-shadow:
    0 0 0 1px #5b9dd9,
    0 0 2px 1px rgba(30, 140, 190, 0.8);
}
```

But not this:

```css
 a {
  box-shadow: 0 0 0 1px #5b9dd9,
    0 0 2px 1px rgba(30, 140, 190, 0.8);
}   
```

I think what you're after is a new rule that limits when a declaration _can_ be multi-line. In this case, it looks at whether the _value_ of the declaration is multi-line or not to determine that.

[Min and Max rules](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/about-rules.md#max-and-min-rules) are used to apply limits. So, perhaps a `declaration-multi-line-value-min-lines`. Where `declaration-multi-line-value` is the _thing_ being targeted and the `min-lines` is what is checked. 

So, `"declaration-multi-line-value-min-lines: 2"` would allow:

```css
 a {
  box-shadow: 0 0 0 1px #5b9dd9,
    0 0 2px 1px rgba(30, 140, 190, 0.8);
}   
```

And 

```css
a {
  box-shadow:
    0 0 0 1px #5b9dd9,
    0 0 2px 1px rgba(30, 140, 190, 0.8);
}
```

But not this:

```css
 a {
  box-shadow:
    0 0 0 1px #5b9dd9;
}   
```

However, there are probably some edge cases I'm not considering. Therefore, I recommend building it [as a plugin](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) first. Experimenting with rules is one of the reasons the plugin system was built.

**Comment by ArmorDarks** [comment]
You are right, it indeed seems like intended but quite unexpected behavior, those this issue should be qualified as a feature request.

> Min and Max rules are used to apply limits. So, perhaps a declaration-multi-line-value-min-lines. Where declaration-multi-line-value is the thing being targeted and the min-lines is what is checked.

Well, that is unexpected thinking, but theoretically should allow even more precise configuration. Though, I'd stick with additional `declaration-colon-newline-after` option, which will forbid newlines for single-valued properties, that way it would be easier to understand what going on.

Maybe it worth at least to keep issue open, just in case someone with experience would like to implement it?
